### PR TITLE
update OMZ 2021.4 documentation

### PR DESCRIPTION
**Pull request by vladimir-dudnik**
update OMZ 2021.4 docs

- [x] update links in root READE.md 
- [x] update root demo README.md (move table with models per device support to models subfolders, remove FPGA)
- [x] add device support table and refer it from Intel/public index.md
- [x] update each demo README.md (add explicit list of supported models, link to the table for model's supported devices, remove note about myriad for some models)

### CODE REVIEW

**Review by andrew-zaytsev — CHANGES_REQUESTED** [review]

File: demos/bert_question_answering_demo/python/README.md [hunk 1]
```diff
@@ -8,14 +8,38 @@ Upon the start-up the demo application reads command line parameters and loads a
 It also fetch data from the user-provided url to populate the "context" text.
 The text is then used to search answers for user-provided questions.
 
+## Preparing to run
+
+Pre-trained models, supported by demo listed in [models.lst](./models.lst) file, located at each demo folder.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_question_answering_demo/python/README.md:13]: instead providing pseudo path `<omz_dir>/demos/bert_question_answering_demo` can we just remove link and rely on what we said at the end of phrase `located at each demo folder`

File: models/intel/index.md [hunk 2]
```diff
@@ -295,4 +296,5 @@ Deep Learning models for speech synthesis (mel spectrogram generation and wave f
 |   text-to-speech-en-multi-0001-generation                          | 48.38 | 12.77 |
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply models/intel/index.md:296]: added refs to demos, downloader and public models

File: demos/README.md [hunk 3]
```diff
@@ -320,4 +252,7 @@ demo, read the demo documentation by clicking the demo name in the demos
 list above.
 
 ## See Also
-* [Introduction to Intel's Deep Learning Inference Engine](https://docs.openvinotoolkit.org/latest/_docs_IE_DG_Introduction.html)
+* [Intel OpenVINO Documentation](https://docs.openvinotoolkit.org/latest/documentation.html)
+
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/README.md:259]: added refs to intel and public models

File: demos/3d_segmentation_demo/python/README.md [hunk 4]
```diff
@@ -8,13 +8,29 @@ Upon the start-up, the demo reads command-line parameters and loads a network an
 
 > **NOTE**: By default, Open Model Zoo demos expect input with BGR channels order. If you trained your model to work with RGB order, you need to manually rearrange the default channels order in the demo application or reconvert your model using the Model Optimizer tool with `--reverse_input_channels` argument specified. For more information about the argument, refer to **When to Reverse Input Channels** section of [Converting a Model Using General Conversion Parameters](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_convert_model_Converting_Model_General.html).
 
+## Preparing to run
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/3d_segmentation_demo/python/README.md]: done

File: demos/bert_question_answering_embedding_demo/python/README.md [hunk 5]
```diff
@@ -17,9 +17,29 @@ Notice that question is usually much shorter than the contexts, so calculating t
 
 If second (conventional SQuAD-tuned) Bert model is provided as well, it is used to further search for the exact answer in the best contexts found in the first step, and the result then also displayed to the user.
 
+## Preparing to run
+
+Pre-trained models, supported by demo listed in [models.lst](./models.lst) file, located at each demo folder.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_question_answering_embedding_demo/python/README.md]: fixed

File: demos/bert_named_entity_recognition_demo/python/README.md [hunk 6]
```diff
@@ -8,14 +8,21 @@ Upon the start-up the demo application reads command line parameters and loads a
 It also fetch data from the user-provided url to populate the "context" text.
 The text is then used to search named entities.
 
+## Preparing to run
+
+Pre-trained models, supported by demo listed in [models.lst](./models.lst) file, located at each demo folder.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_named_entity_recognition_demo/python/README.md]: fixed

File: demos/object_detection_demo/cpp/README.md [hunk 7]
```diff
@@ -97,15 +157,12 @@ Options:
...
-
-> **NOTE**: Before running the demo with a trained model, make sure the model is converted to the Inference Engine format (\*.xml + \*.bin) using the [Model Optimizer tool](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html).
-
-If labels file is used, it should correspond to model output. Demo suggests labels listed in the file to be indexed from 0, one line - one label (i.e. very first line contains label for ID 0). Note that some models may return labels IDs in range 1..N, in this case label file should contain "background" label at the very first line.
+If labels file is used, it should correspond to model output. Demo treat labels, listed in the file, to be indexed from 0, one line - one label (i.e. very first line contains label for ID 0). Note that some models may return labels IDs in range 1..N, in this case label file should contain "background" label at the very first line.
```
Comment (tsavina) [inline demos/object_detection_demo/cpp/README.md:160]: i.e. ---> that is

File: demos/README.md [hunk 8]
```diff
@@ -62,77 +62,9 @@ To run the demo applications, you can use images and videos from the media files
 
 ## Demos that Support Pre-Trained Models
 
-> **NOTE:** Inference Engine HDDL and FPGA plugins are available in [proprietary](https://software.intel.com/en-us/openvino-toolkit) distribution only.
+> **NOTE:** Inference Engine HDDL plugin are available in [proprietary](https://software.intel.com/en-us/openvino-toolkit) distribution only.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/README.md]: done

File: demos/bert_question_answering_demo/python/README.md [hunk 9]
```diff
@@ -8,14 +8,38 @@ Upon the start-up the demo application reads command line parameters and loads a
...
+
+* bert-large-uncased-whole-word-masking-squad-0001
+* bert-large-uncased-whole-word-masking-squad-emb-0001
+* bert-large-uncased-whole-word-masking-squad-int8-0001
+* bert-small-uncased-whole-word-masking-squad-0001
+* bert-small-uncased-whole-word-masking-squad-0002
+* bert-small-uncased-whole-word-masking-squad-emb-int8-0001
+* bert-small-uncased-whole-word-masking-squad-int8-0002
+
+The "small" variants of these are so called "distilled" models, which originated from the BERT Large but substantially smaller and faster.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_question_answering_demo/python/README.md]: done

File: demos/bert_question_answering_embedding_demo/python/README.md [hunk 10]
```diff
@@ -108,23 +104,41 @@ You can use the following command to try the demo (assuming the model from the O
...
+
 The demo will use the Wikipedia articles about the Bert character and the speed of light to answer your questions like
 "what is the speed of light", "how to measure the speed of light", "who is Bert", "how old is Bert", etc.
 
+## Demo Inputs
+
+The application reads text from the HTML pages at the given urls and then answers questions typed from the console.
+The models and its parameters (inputs and outputs) are also important demo arguments.
+Notice that since order of inputs for the model does matter, the demo script checks that the inputs specified
+from the command-line match the actual network inputs.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_question_answering_embedding_demo/python/README.md]: done

File: demos/bert_question_answering_embedding_demo/python/README.md [hunk 11]
```diff
@@ -108,23 +104,41 @@ You can use the following command to try the demo (assuming the model from the O
...
+The embedding model is reshaped by the demo to infer embedding vectors for long contexts and short question.
+Be sure that the original model converted by Model Optimizer with reshape option.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/bert_question_answering_embedding_demo/python/README.md]: done

File: demos/classification_demo/cpp/README.md [hunk 12]
```diff
@@ -51,17 +156,14 @@ Options:
...
+The number of `InferRequest`s is specified by -nireq flag. Each `InferRequest` acts as a "buffer": it waits in queue before being filled with images and sent for inference, then after the inference completes, it waits in queue until its results are processed. Increasing the number of `InferRequest`s usually increases performance, because in that case multiple `InferRequest`s can be processed simultaneously if the device supports parallelization. However, big number of `InferRequest`s increases latency because each image still needs to wait in queue.
 
-> **NOTE**: Before running the demo with a trained model, make sure the model is converted to the Inference Engine format (\*.xml + \*.bin) using the [Model Optimizer tool](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html).
+For higher FPS, using `-nireq` which slightly exceeds `-nstreams` value summed over all used devices is recommended.
 
 For example, use the following command line command to run the application:
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/classification_demo/cpp/README.md]: done

File: demos/face_detection_mtcnn_demo/python/README.md [hunk 13]
```diff
@@ -2,27 +2,30 @@
...
 
-```
-python3 face_detection_mtcnn_demo.py -h
-```
+> **NOTE**: Refer to the tables [Intel's Pre-Trained Models Device Support](../../../models/intel/device_support.md) and [Public Pre-Trained Models Device Support](../../../models/public/device_support.md) for the details on models inference support at different devices.
 
-The command yields the following usage message:
+## Running
+
+Run the application with the `-h` option to see available command line parameters:
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/face_detection_mtcnn_demo/python/README.md]: done

File: demos/formula_recognition_demo/python/README.md [hunk 14]
```diff
@@ -60,41 +66,57 @@ MacOS:
...
 2. Image from the input target will be binarized, preprocessed and fed to the network. Preprocessed and binarized image is placed on the top of the window (near `Model input` label)
 3. If the formula will be predicted with sufficient confidence score, it will be placed right under preprocessed image (near `Predicted` label)
 4. If rendering is available (see the previous Paragraph for details) and predicted formula does not contain LaTeX grammar errors, it will be rendered and placed near `Rendered` label.
 
 Navigation keys:
-  * Use `q` button to quit from program
-  * Use `o` to decrease the size of the input (red) window
-  * Use `p` to increase the size of the input window
+
+* Use `q` button to quit from program
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/formula_recognition_demo/python/README.md]: done

File: demos/handwritten_text_recognition_demo/python/README.md [hunk 15]
```diff
@@ -61,27 +66,28 @@ Options:
...
 
-For example, if we want to restrict the output characters to only digits and hyphens, we need to provide the path to the designated character file, e.g. `digit_hyphen.txt`. Then the script will perform a post-filtering processing on the output characters, but please note that it is possible that other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements.
+For example, if we want to restrict the output characters to only digits and hyphens, we need to provide the path to the designated character file, e.g. `digit_hyphen.txt`. Then the script will perform a post-filtering processing on the output characters, but please note that it is possible that other characters are still allowed if none of `digit_hyphen.txt` is in first K chosen elements. The mentioned characters text files located at `data` subfolder of this demo.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/handwritten_text_recognition_demo/python/README.md]: done

File: demos/human_pose_estimation_3d_demo/python/README.md [hunk 16]
```diff
@@ -24,10 +22,23 @@ The demo workflow is the following:
...
+
+* human-pose-estimation-3d-0001
+
+> **NOTE**: Refer to the tables [Intel's Pre-Trained Models Device Support](../../../models/intel/device_support.md) and [Public Pre-Trained Models Device Support](../../../models/public/device_support.md) for the details on models inference support at different devices.
+
+### Prerequisites
 
 This demo application requires a native Python extension module to be built before you can run it.
-Refer to [Using Open Model Zoo demos](../../README.md), for instructions on how to build it and prepare the environment for running the demo.
+Refer to [Open Model Zoo Demos](../../README.md), for instructions on how to build it and prepare the environment for running the demo.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/human_pose_estimation_3d_demo/python/README.md]: done

File: demos/interactive_face_detection_demo/cpp/README.md [hunk 17]
```diff
@@ -1,40 +1,71 @@
...
-1.	The application reads command-line parameters and loads up to five networks depending on `-m...` options family to the Inference
+1. The application reads command-line parameters and loads up to five networks depending on `-m...` options family to the Inference
 Engine.
-2.	The application gets a frame from the OpenCV VideoCapture.
-3.	The application performs inference on the Face Detection network.
-4.	The application performs up to five simultaneous inferences, using the Age/Gender, Head Pose, Emotions, Facial Landmarks and Anti-spoof detection networks if they are specified in the command line.
-5.	The application displays the results.
+2. The application gets a frame from the OpenCV VideoCapture.
+3. The application performs inference on the Face Detection network.
+4. The application performs up to five simultaneous inferences, using the Age/Gender, Head Pose, Emotions, Facial Landmarks and Anti-spoof detection networks if they are specified in the command line.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/interactive_face_detection_demo/cpp/README.md]: done

File: demos/interactive_face_detection_demo/cpp_gapi/README.md [hunk 18]
```diff
@@ -3,33 +3,63 @@
...
 
-1.	The application reads command-line parameters and loads up to five networks depending on `-m...` options family to the Inference
+1. The application reads command-line parameters and loads up to five networks depending on `-m...` options family to the Inference
 Engine.
-2.	G-API pipeline gets a video/image/camera like a source using the OpenCV VideoCapture.
-3.	G-API pipeline performs inference on the Face Detection network.
-4.	G-API pipeline runs post processing kernel.
-5.	G-API pipeline performs four inferences, using the Age/Gender, Head Pose, Emotions, and Facial Landmarks detection networks if they are specified in the command line.
-6.	The application displays the results.
+2. G-API pipeline gets a video/image/camera like a source using the OpenCV VideoCapture.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/interactive_face_detection_demo/cpp_gapi/README.md]: done

File: demos/multi_channel_object_detection_demo_yolov3/cpp/README.md [hunk 19]
```diff
@@ -1,26 +1,35 @@
 # Multi-Channel Object Detection Yolov3 C++ Demo
 
-This demo provides an inference pipeline for multi-channel yolo v3. The demo uses Yolo v3 Object Detection network. You can follow [this](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_YOLO_From_Tensorflow.html) page convert the YOLO V3 and tiny YOLO V3 into IR model and execute this demo with converted IR model.
-
-> **NOTES**:
-> If you don't use [this](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_YOLO_From_Tensorflow.html) page to convert the model, it may not work.
+This demo provides an multi channel inference pipeline for YOLO v3 Object Detection network.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/multi_channel_object_detection_demo_yolov3/cpp/README.md]: done

File: demos/object_detection_demo/cpp/README.md [hunk 20]
```diff
@@ -29,8 +27,7 @@ need to pull Inference Engine demos helpers to your app
 
 ## How It Works
 
-On the start-up, the application reads command line parameters and loads a network to the Inference
-Engine. Upon getting a frame from the OpenCV VideoCapture it performs inference and displays the results.
+On the start-up, the application reads command line parameters and loads a network to the Inference Engine. Upon getting a frame from the OpenCV VideoCapture it performs inference and displays the results.
```
[the remark this answers is not in the document]
Comment (vladimir-dudnik) [reply demos/object_detection_demo/cpp/README.md]: done

File: demos/object_detection_demo/cpp/README.md (hunk 7 above)
Comment (vladimir-dudnik) [reply demos/object_detection_demo/cpp/README.md]: done

**Review by tsavina — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Pull request merged**